worker/pkg/workflows/datasync/activities/gen-benthos-configs/query-builder.go

```diff
@@ -0,0 +1,35 @@
+package genbenthosconfigs_activity
+
+import (
+	"fmt"
+
+	"github.com/doug-martin/goqu/v9"
+	// import the dialect
+	_ "github.com/doug-martin/goqu/v9/dialect/mysql"
+	_ "github.com/doug-martin/goqu/v9/dialect/postgres"
+)
+
+func buildSelectQuery(
+	driver, schema, table string,
+	columns []string,
+	whereClause *string,
+) (string, error) {
+	builder := goqu.Dialect(driver)
+	sqltable := goqu.S(schema).Table(table)
+
+	selectColumns := make([]any, len(columns))
+	for i, col := range columns {
+		selectColumns[i] = col
+	}
+	query := builder.From(sqltable).Select(selectColumns...)
+
+	if whereClause != nil && *whereClause != "" {
+		query = query.Where(goqu.L(*whereClause))
+	}
+	sql, _, err := query.Prepared(true).ToSQL()
```

Comment:
we will eventually want to return the parameters here. however, the user provides the where statment today so we'd need to run logic over it to parameterize it before we pass it in to Goqu, as goqu only parameterizes if you provide a Where sturct instead of the already stringified version.